### DISCUSSION TITLE
Renovate missing updates in description table

### DISCUSSION THREAD

**Discussion by dabljues** [renovatebot/renovate#32102] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab, version 38

### Please tell us more about your question or problem

I have a monorepo that has multiple services. I have some dependencies that have to be bumped in multiple services in sync (for example Python's `celery` or `redis` packages). This is to ensure that services are working on the same version.

To achieve that I have a group like this:

```json
{
    "matchUpdateTypes": [
        "patch",
        "minor"
    ],
    "groupName": "celery bump",
    "matchFileNames": [
        "foo/pyproject.toml",
        "bar/pyproject.toml",
        "baz/pyproject.toml"
    ],
    "matchPackageNames": [
        "celery"
    ],
    "additionalBranchPrefix": "{{{depNameSanitized}}}-",
    "prBodyDefinitions": {
        "Package": "{{{depNameLinked}}} ({{{packageFileDir}}})"
    },
    "group": {
        "branchTopic": "workers",
        "commitMessageTopic": "dependency {{{depName}}}"
    }
}
```

Now, everything works as expected. I get a single MR that bumps the `celery` dependency in those 3 services. However, there's a problem. The "changes" table in the description, instead of having three rows (3 services), has one, like this:

| Package | Package file | Type | Update | Change |
|---|---|---|---|---|
| celery (foo) | foo/pyproject.toml | project.dependencies | minor | `==0.0.1` -> `==0.1.0` |

Instead it should be 3 rows, like this:

| Package | Package file | Type | Update | Change |
|---|---|---|---|---|
| celery (foo) | foo/pyproject.toml | project.dependencies | minor | `==0.0.1` -> `==0.1.0` |
| celery (bar) | bar/pyproject.toml | project.dependencies | minor | `==0.0.1` -> `==0.1.0` |
| celery (baz) | baz/pyproject.toml | project.dependencies | minor | `==0.0.1` -> `==0.1.0` |

The issue is, in my opinion, that the ["uniqueness" key](https://github.com/renovatebot/renovate/blob/main/lib/workers/repository/update/pr/index.ts#L259) for the upgrades doesn't take the "Package" nor "Package file" into consideration. Similar issue was described here: https://github.com/renovatebot/renovate/discussions/26964.

The problem only persists if the upgrades in those services are from `X` version to `Y` version. If one of the services is bumped to a newer version for some reason, or is bumped **from** an older one, it will appear in the table, because it's unique. 

Can this be addressed? I know that this is not the end of the world, but it's misleading. In my case, I will be disabling the PR table, because it's better not to have it that have it provide false information. I think that I'm not the only one working in a monorepo/grouping multi-service/multi-directory bumps like this,, so I think that a change like this would be welcome by more people.

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
This won't be addressed by the core team, but you're welcome to look into it further and propose a solution if you think one exists. Deduping based on file name is definitely the right default behavior for the majority. For others, the ideal would be to use the PR body templating capabilities to override the defaults

**Reply by dabljues** [reply]
Yeah, the config option would probably the best. I'm not sure whether my JS skills will be enough to introduce this change in a way that would be up to your standards, but if I'll feel like I can do it, I will provide it
